tutorials/tts/Audio_Codec_Training.ipynb

```diff
@@ -0,0 +1,667 @@
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "ai0zbXSOriuY"
+      },
+      "outputs": [],
+      "source": [
+        "preprocess_audio(\"dev\")"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "NUKnidQYfgDo"
+      },
+      "outputs": [],
+      "source": [
+        "preprocess_audio(\"train\")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "x2yhJtsj2lDR"
+      },
+      "source": [
+        "We should listen to a few audio files before and after the processing so be sure we configured it correctly.\n",
+        "\n",
+        "Note that the processed audio is shorter because we trimmed the leading and trailing silence."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "_fM3GwJxkjOA"
+      },
+      "outputs": [],
+      "source": [
+        "audio_file = \"p228_009.wav\"\n",
+        "audio_filepath = input_audio_dir / audio_file\n",
+        "processed_audio_filepath = output_audio_dir / audio_file.replace(\".wav\", \".flac\")\n",
+        "\n",
+        "print(\"Original audio.\")\n",
+        "ipd.display(ipd.Audio(audio_filepath))\n",
+        "\n",
+        "print(\"Processed audio.\")\n",
+        "ipd.display(ipd.Audio(processed_audio_filepath))"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "oRO842MUyODC"
+      },
+      "source": [
+        "# Audio Codec Training"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "E4wUKYOfH8ax"
+      },
+      "source": [
+        "Here we show how to train an audio codec model from scratch. Instructions and checkpoints for fine-tuning will be provided later.\n"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "pqfl9jAYMJob"
+      },
+      "outputs": [],
+      "source": [
+        "import torch\n",
+        "from omegaconf import OmegaConf"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "jK2rr-Kr6Qg8"
+      },
+      "outputs": [],
+      "source": [
+        "dataset_name = \"vctk\"\n",
+        "audio_dir = DATA_DIR / \"audio_preprocessed\"\n",
+        "train_manifest_filepath = DATA_DIR / \"train_manifest.json\"\n",
+        "dev_manifest_filepath = DATA_DIR / \"dev_manifest.json\""
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "Vr4D-NB-yQx8"
+      },
+      "outputs": [],
+      "source": [
+        "audio_codec_training_script = NEMO_EXAMPLES_DIR / \"audio_codec.py\"\n",
+        "\n",
+        "# The total number of training steps will be (epochs * steps_per_epoch)\n",
+        "epochs = 10\n",
+        "steps_per_epoch = 10\n",
+        "\n",
```

Comment:
I feel this tutorial would be even more helpful with some info on Audio Codec vs MelCodec differences and explanation of main parameters on how we could switch from RVQ to FSQ and how to control bitrate

Reply:
I am thinking that level of detail would be better left for a separate tutorial or primer document. Without better supporting documentation/papers, no one would understand it right now unless they are already very familiar with codec research.

Reply:
what would be separate tutorial?

Reply:
First we are missing a high-level primer for the domain where we can explain with visuals/examples what high level codec concepts are, like bitrate, quantizers, etc. That is, the codec equivalent of https://github.com/NVIDIA/NeMo/blob/main/tutorials/00_NeMo_Primer.ipynb, https://github.com/NVIDIA/NeMo/blob/main/tutorials/tts/NeMo_TTS_Primer.ipynb, https://github.com/NVIDIA/NeMo/blob/main/tutorials/asr/ASR_with_NeMo.ipynb

There might be some information we should include here, its just hard to say without having the supporting documentation already in place. In general I am not sure if architecture overrides should be in a jupyter notebook or some other form of documentation since we expect users to override model definition primarily by modifying the .yaml file directly, not by overriding it in the CLI.

Would be happy to hear others thoughts & opinions.

Reply:
Have you chosen not to include details on bitrates, the types of codecs available, and how we can adjust the parameters to obtain various bitrates and codec types?

Reply:
Except this everything LGTM!

Reply:
I think additional information can be added later when we have more than 1 codec available